python312Packages.bite-parser: 0.2.4 -> 0.2.5

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for python312Packages.bite-parser is: Asynchronous parser taking incremental bites out of your byte input stream

meta.homepage for python312Packages.bite-parser is: https://github.com/jgosmann/bite-parser

meta.changelog for python312Packages.bite-parser is: https://github.com/jgosmann/bite-parser/blob/v0.2.5/CHANGELOG.rst


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0.2.5 with grep in /nix/store/flbfq4mz5mlyfjp9qwpzdl8kg9a6f70a-python3.12-bite-parser-0.2.5
- found 0.2.5 in filename of file in /nix/store/flbfq4mz5mlyfjp9qwpzdl8kg9a6f70a-python3.12-bite-parser-0.2.5

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
4 total rebuild path(s)

4 package rebuild(s)

First fifty rebuilds by attrpath

dmarc-metrics-exporter
python311Packages.bite-parser
python312Packages.bite-parser
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/flbfq4mz5mlyfjp9qwpzdl8kg9a6f70a-python3.12-bite-parser-0.2.5 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A python312Packages.bite-parser https://github.com/r-ryantm/nixpkgs/archive/9ba8bd718610a347f7cf8558b46dc5dc96b7016f.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/9ba8bd718610a347f7cf8558b46dc5dc96b7016f#python312Packages.bite-parser
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/flbfq4mz5mlyfjp9qwpzdl8kg9a6f70a-python3.12-bite-parser-0.2.5
ls -la /nix/store/flbfq4mz5mlyfjp9qwpzdl8kg9a6f70a-python3.12-bite-parser-0.2.5/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>dmarc-metrics-exporter</li>
    <li>dmarc-metrics-exporter.dist</li>
    <li>python311Packages.bite-parser</li>
    <li>python311Packages.bite-parser.dist</li>
    <li>python312Packages.bite-parser</li>
    <li>python312Packages.bite-parser.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @dotlambda for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dotlambda commented on pkgs/development/python-modules/bite-parser/default.nix:
```diff
@@ -19,7 +19,7 @@ buildPythonPackage rec {
+    hash = "sha256-ROGVhvCZUBo0PqNRQWnQ++Sug6LZe6kgw4SSzf+7QqE=";
   };
 
   nativeBuildInputs = [ poetry-core ];
```
```suggestion
  build-system = [ poetry-core ];
```

dotlambda commented on pkgs/development/python-modules/bite-parser/default.nix:
```diff
@@ -10,7 +10,7 @@
-  version = "0.2.4";
+  version = "0.2.5";
 
   disabled = pythonOlder "3.8";
```
```suggestion
  disabled = pythonOlder "3.9";
```

dotlambda reviewed (commented):
Use `pyproject = true`.

dotlambda committed:
python312Packages.bite-parser: 0.2.4 -> 0.2.5

Diff: https://github.com/jgosmann/bite-parser/compare/refs/tags/v0.2.4...v0.2.5

Changelog: https://github.com/jgosmann/bite-parser/blob/refs/tags/v0.2.5/CHANGELOG.rst

dotlambda commented:
Result of `nixpkgs-review pr 351686` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>dmarc-metrics-exporter</li>
    <li>dmarc-metrics-exporter.dist</li>
    <li>python311Packages.bite-parser</li>
    <li>python311Packages.bite-parser.dist</li>
    <li>python312Packages.bite-parser</li>
    <li>python312Packages.bite-parser.dist</li>
  </ul>
</details>
